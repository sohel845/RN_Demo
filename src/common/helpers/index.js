export * from './test-id-helper'
export * from './test-id-prop-type'
export * from './mock-store'
