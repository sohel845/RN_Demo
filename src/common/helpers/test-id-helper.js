export function getTestIdProps (id) {
    // TODO: strip in PROD builds
    return { accessible: true, accessibilityLabel: id, testID: id }
}
