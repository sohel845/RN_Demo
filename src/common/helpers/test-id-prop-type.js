import PropTypes from 'prop-types'

export const testIdPropType = PropTypes.shape({
    accessibilityLabel: PropTypes.string,
    accessible: PropTypes.bool,
    testID: PropTypes.string
})
