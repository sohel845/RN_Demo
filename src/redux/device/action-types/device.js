export const SET_DEVICE_ORIENTATION = 'device/SET_DEVICE_ORIENTATION'
export const SET_DEVICE_TYPE = 'device/SET_DEVICE_TYPE'
