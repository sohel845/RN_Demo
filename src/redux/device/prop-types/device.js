import PropTypes from 'prop-types'
import { DEVICE_TYPE, DEVICE_ORIENTATION } from '../constants'

export const devicePropType = PropTypes.shape({
    orientation: PropTypes.oneOf(Object.values(DEVICE_ORIENTATION)).isRequired,
    type: PropTypes.oneOf(Object.values(DEVICE_TYPE)).isRequired
})
