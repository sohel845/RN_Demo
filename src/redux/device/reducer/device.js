import {
    SET_DEVICE_ORIENTATION,
    SET_DEVICE_TYPE
} from '../action-types'
import { DEVICE_TYPE, DEVICE_ORIENTATION } from '../constants'

export const INITIAL_STATE = {
    type: DEVICE_TYPE.TABLET,
    orientation: DEVICE_ORIENTATION.LANDSCAPE
}

export default function reducer (device = INITIAL_STATE, action = {}) {
    switch (action.type) {
    case SET_DEVICE_ORIENTATION: {
        const { orientation } = action
        if (Object.values(DEVICE_ORIENTATION).includes(orientation)) {
            console.log('device orientation set', orientation)
            return Object.assign({}, device, {
                orientation: orientation
            })
        }
        return device
    }
    case SET_DEVICE_TYPE: {
        const { deviceType } = action
        if (Object.values(DEVICE_TYPE).includes(deviceType)) {
            console.log('device type set', deviceType)
            return Object.assign({}, device, {
                type: deviceType
            })
        }
        return device
    }
    default:
        return device
    }
}
