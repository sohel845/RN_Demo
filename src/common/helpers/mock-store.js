import React from 'react'
import { createStore, combineReducers } from 'redux'
import { Provider } from 'react-redux'
import { reducer as themeReducer } from '../../redux/themes'
import { reducer as deviceReducer } from '../../redux/device'
import { create } from 'react-test-renderer'
import { mockTheme } from '../../redux/themes/__mocks__/themes'
import { DEVICE_VARIANT } from '../../redux/device'

export function createThemedMockStore (device = DEVICE_VARIANT.PHONE_PORTRAIT, extraReducers = {}, extraStore = {}) {
    return createStore(combineReducers({
        theme: themeReducer,
        device: deviceReducer,
        ...extraReducers
    }), {
        theme: { name: 'cyan' },
        device: device,

        ...extraStore })
}
