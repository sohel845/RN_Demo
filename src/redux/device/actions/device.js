import { SET_DEVICE_ORIENTATION, SET_DEVICE_TYPE } from '../action-types'

export function setDeviceOrientation (orientation) {
    return { type: SET_DEVICE_ORIENTATION, orientation }
}

export function setDeviceType (deviceType) {
    return { type: SET_DEVICE_TYPE, deviceType }
}
