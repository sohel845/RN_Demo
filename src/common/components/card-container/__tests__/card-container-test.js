import renderer from 'react-test-renderer'
import React from 'react'
import { Provider } from 'react-redux'

import CardContainer from '../card-container'
import Card from '../../card/card'
import CircleProgressBar from '../../circle-progress-bar/circle-progress-bar'
import Badge, { BADGE_SIZE } from '../../badge/badge'
import { createThemedMockStore } from '../../../helpers'
import { DEVICE_VARIANT } from '../../../../redux/device'

describe('Card Component', () => {
    beforeEach(() => {
        emptyStore = createThemedMockStore(DEVICE_VARIANT.PHONE_PORTRAIT)
    })

    const createElement = (props) => {
        return renderer.create(<Provider {...{ store: emptyStore }} ><CardContainer {...props}>
            <Card
                numColumns={1}
                label={'Label'}
                progressBar={<CircleProgressBar
                    percent={80}
                    radius={40} />}
                badge={<Badge size={BADGE_SIZE.L} content={'1'} />}
                fixHeight={true}
                text={'Since 11:45'} />
        </CardContainer></Provider>)
    }

    it('renders correctly', () => {
        const element = createElement()
        expect(element.toJSON()).toMatchSnapshot()
    })
})

