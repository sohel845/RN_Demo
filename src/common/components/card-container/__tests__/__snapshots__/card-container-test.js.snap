// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
      },
      Object {
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00000066",
        "flexDirection": "row",
        "height": 133,
        "opacity": 1,
        "overflow": "hidden",
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 120,
          "justifyContent": "center",
          "width": 120,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 5000,
                "padding": 3,
              },
              Object {
                "backgroundColor": "#333D46",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#748b9f",
                  "borderRadius": 40,
                  "height": 80,
                  "width": 80,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "zIndex": 1,
                  },
                  Object {
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                  },
                  Object {
                    "height": 80,
                    "width": 40,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#B6D345",
                    "borderBottomRightRadius": 0,
                    "borderRadius": 40,
                    "borderTopRightRadius": 0,
                    "height": 80,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 20,
                      },
                      Object {
                        "rotate": "0deg",
                      },
                      Object {
                        "translateX": -20,
                      },
                    ],
                    "width": 40,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "zIndex": 3,
                  },
                  Object {
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                  },
                  Object {
                    "height": 80,
                    "width": 40,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#748b9f",
                    "borderBottomRightRadius": 0,
                    "borderRadius": 40,
                    "borderTopRightRadius": 0,
                    "height": 80,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 20,
                      },
                      Object {
                        "rotate": "0deg",
                      },
                      Object {
                        "translateX": -20,
                      },
                    ],
                    "width": 40,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "zIndex": 2,
                  },
                  Object {
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                  },
                  Object {
                    "height": 80,
                    "width": 40,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#B6D345",
                    "borderBottomRightRadius": 0,
                    "borderRadius": 40,
                    "borderTopRightRadius": 0,
                    "height": 80,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 20,
                      },
                      Object {
                        "rotate": "0deg",
                      },
                      Object {
                        "translateX": -20,
                      },
                    ],
                    "width": 40,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "#333D46",
                    "borderRadius": 37,
                    "height": 74,
                    "width": 74,
                    "zIndex": 4,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "paddingVertical": 20,
          "position": "relative",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#A2AAAF",
            "fontFamily": "FrutigerNextCondensedHeavy",
            "fontSize": 14.5,
          }
        }
      >
        Label
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "FrutigerNextCondensedHeavy",
              "fontSize": 25,
              "marginRight": 10,
            }
          }
        >
          Since 11:45
        </Text>
        <BVLinearGradient
          angle={120}
          angleCenter={
            Object {
              "x": 0,
              "y": 1,
            }
          }
          colors={
            Array [
              4288850607,
              4285825160,
            ]
          }
          endPoint={
            Object {
              "x": 0.5,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0.5,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 0,
                  "justifyContent": "center",
                  "position": "relative",
                },
                Object {
                  "borderColor": "#74808899",
                },
                Object {
                  "borderRadius": 18,
                  "borderWidth": 3.5,
                  "height": 36,
                  "minWidth": 29,
                },
              ],
            ]
          }
          useAngle={true}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "flex": 0,
                  "position": "relative",
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "FrutigerNextCondensedRegular",
                  "fontSize": 18,
                  "marginHorizontal": 7,
                  "minWidth": 14.5,
                },
              ]
            }
          >
            1
          </Text>
        </BVLinearGradient>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#748088",
            "bottom": -10,
            "height": 1,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</View>
`;
